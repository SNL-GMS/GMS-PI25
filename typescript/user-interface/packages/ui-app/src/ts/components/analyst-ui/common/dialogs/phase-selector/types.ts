export interface CategorizedPhases {
  categoryTitle: string;
  phases: string[];
}

export interface PhaseList {
  listTitle: string;
  categorizedPhases: CategorizedPhases[];
}
