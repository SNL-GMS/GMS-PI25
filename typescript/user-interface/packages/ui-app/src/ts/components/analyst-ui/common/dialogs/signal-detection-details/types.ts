import type { SignalDetectionTypes } from '@gms/common-model';
import type { Row } from '@gms/ui-core-components';

/**
 * SignalDetectionDetails Props
 */
export interface SignalDetectionDetailsProps {
  signalDetection: SignalDetectionTypes.SignalDetection;
}

export interface SignalDetectionDetailsData {
  signalDetection: SignalDetectionTypes.SignalDetection;
  color: string;
  assocStatus: string;
  swatchTooltipText: string;
}

/**
 * Interface that describes the Detection history
 * information.
 */
export interface SignalDetectionHistoryRow extends Row {
  id: string;
  versionId: string;
  creationTimestamp: number;
  phase: string;
  arrivalTimeMeasurementFeatureType: string;
  arrivalTimeMeasurementTimestamp: number;
  arrivalTimeMeasurementUncertaintySec: number;
  channelSegmentType: string;
  authorName: string;
  deleted: string;
}
