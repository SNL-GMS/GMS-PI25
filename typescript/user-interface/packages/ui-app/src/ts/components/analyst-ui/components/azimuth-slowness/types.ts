import type {
  ChannelSegmentTypes,
  ColorTypes,
  CommonTypes,
  ConfigurationTypes,
  EventTypes,
  FkTypes,
  SignalDetectionTypes,
  StationTypes,
  WorkflowTypes
} from '@gms/common-model';
import type GoldenLayout from '@gms/golden-layout';
import type {
  AnalystWorkspaceTypes,
  AppDispatch,
  ChannelSegmentFetchResult,
  EventsFetchResult,
  EventStatus,
  FindEventStatusInfoByStageIdAndEventIdsQuery,
  FkChannelSegmentRecord,
  FkFrequencyThumbnailRecord,
  SignalDetectionFetchResult,
  StationQueryProps,
  UIChannelSegmentRecord
} from '@gms/ui-state';
import type Immutable from 'immutable';

import type {
  FilterType,
  FkThumbnailSize
} from './components/fk-thumbnail-list/fk-thumbnails-controls';

/**
 * Used to return a super set of the fk configuration from the fk config popover
 */
export type FkConfigurationWithUnits = FkTypes.FkConfiguration & {
  fkUnitToDisplay: FkTypes.FkUnits;
};

/**
 * Azimuth Slowness Redux Props
 */
export interface AzimuthSlownessReduxProps {
  // passed in from golden-layout
  glContainer?: GoldenLayout.Container;
  viewableInterval: CommonTypes.TimeRange;
  openEventId: string;
  sdIdsToShowFk: string[];
  analysisMode: WorkflowTypes.AnalysisMode;
  location: AnalystWorkspaceTypes.LocationSolutionState;
  selectedSortType: AnalystWorkspaceTypes.WaveformSortType;
  setSdIdsToShowFk(signalDetectionIds: string[]): void;
  setMeasurementModeEntries(entries: Record<string, boolean>): void;
  signalDetectionResults: SignalDetectionFetchResult;
  channelSegmentResults: ChannelSegmentFetchResult;
  fkChannelSegments: FkChannelSegmentRecord;
  fkFrequencyThumbnails: FkFrequencyThumbnailRecord;
  eventResults: EventsFetchResult;
  eventStatusQuery: FindEventStatusInfoByStageIdAndEventIdsQuery;
  openIntervalName: string;
  uiTheme: ConfigurationTypes.UITheme;
  colorMap: ColorTypes.ColorMapName;
  dispatch: AppDispatch;
}

export interface SubscriptionAction {
  (
    list: SignalDetectionTypes.SignalDetection[],
    index: number,
    prev: SignalDetectionTypes.SignalDetection[],
    currentIteree: SignalDetectionTypes.SignalDetection
  ): void;
}

/**
 * Azimuth Slowness State
 */
export interface AzimuthSlownessState {
  fkThumbnailSizePx: FkThumbnailSize;
  fkThumbnailColumnSizePx: number;
  filterType: FilterType;
  numberOfOutstandingComputeFkMutations: number;
  fkUnitsForEachSdId: Immutable.Map<string, FkTypes.FkUnits>;
  fkInnerContainerWidthPx: number;
  displayedSignalDetectionId: string;
}

/**
 * Mutations used by the Az Slow display
 */
export interface AzimuthSlownessMutations {
  computeFks: (fkInput: FkTypes.FkInputWithConfiguration[]) => Promise<void>;
  setWindowLead: (args: any) => Promise<void>;
  markFkReviewed(channelSegmentDescriptor: ChannelSegmentTypes.ChannelSegmentDescriptor): void;
}

/**
 * Consolidated props for Azimuth Slowness
 */
export type AzimuthSlownessProps = AzimuthSlownessReduxProps &
  AzimuthSlownessMutations &
  StationQueryProps &
  SignalDetectionFetchResult &
  ChannelSegmentFetchResult &
  EventsFetchResult &
  FindEventStatusInfoByStageIdAndEventIdsQuery;

/**
 * State of the az slow panel
 */
export interface AzimuthSlownessPanelState {
  currentMovieSpectrumIndex: number;
  selectedSdIds: string[];
}

export interface AzimuthSlownessPanelProps {
  // Data
  defaultStations: StationTypes.Station[];
  eventsInTimeRange: EventTypes.Event[];
  eventStatuses: Record<string, EventStatus>;
  displayedSignalDetection: SignalDetectionTypes.SignalDetection | undefined;
  openEvent: EventTypes.Event | undefined;
  associatedSignalDetections: SignalDetectionTypes.SignalDetection[];
  unassociatedSignalDetections: SignalDetectionTypes.SignalDetection[];
  signalDetectionsToDraw: SignalDetectionTypes.SignalDetection[];
  signalDetectionsIdToFeaturePredictions: Immutable.Map<string, EventTypes.FeaturePrediction[]>;
  signalDetectionsByStation: SignalDetectionTypes.SignalDetection[];
  channelSegments: UIChannelSegmentRecord;
  selectedFk: FkTypes.FkPowerSpectra;
  fkChannelSegments: FkChannelSegmentRecord;
  fkFrequencyThumbnails: FkTypes.FkFrequencyThumbnail[];
  featurePredictionsForDisplayedSignalDetection: EventTypes.FeaturePrediction[];
  distances: EventTypes.LocationDistance[];
  location: AnalystWorkspaceTypes.LocationSolutionState;
  // Azimuth display state as props
  fkThumbnailColumnSizePx: number;
  fkDisplayWidthPx: number;
  fkDisplayHeightPx: number;
  filterType: FilterType;
  fkThumbnailSizePx: FkThumbnailSize;
  fkUnitsForEachSdId: Immutable.Map<string, FkTypes.FkUnits>;
  numberOfOutstandingComputeFkMutations: number;
  fkUnitForDisplayedSignalDetection: FkTypes.FkUnits;
  fkInnerContainerWidthPx: number;
  selectedSortType: AnalystWorkspaceTypes.WaveformSortType;
  uiTheme: ConfigurationTypes.UITheme;
  openIntervalName: string;
  colorMap: ColorTypes.ColorMapName;
  // Prop functions
  adjustFkInnerContainerWidth(
    fkThumbnailsContainer: HTMLDivElement,
    fkThumbnailsInnerContainer: HTMLDivElement
  ): void;
  updateFkThumbnailSize(size: FkThumbnailSize): void;
  updateFkFilter(filterType: FilterType): void;
  setFkThumbnailColumnSizePx(newSizePx: number): void;
  computeFkAndUpdateState(
    fkParams: FkTypes.FkParams,
    configuration: FkTypes.FkDialogConfiguration
  ): void;
  setFkUnitForSdId(sdId: string, fkUnit: FkTypes.FkUnits): void;
  setSdIdsToShowFk(signalDetectionIds: string[]): void;
  setDisplayedSignalDetection(sd: SignalDetectionTypes.SignalDetection): void;
  setMeasurementModeEntries(entries: Record<string, boolean>): void;
}
