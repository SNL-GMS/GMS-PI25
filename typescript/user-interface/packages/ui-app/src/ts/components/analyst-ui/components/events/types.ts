import type { EventTypes } from '@gms/common-model';
import type { CellRendererParams, Row } from '@gms/ui-core-components';
import type { ArrivalTime } from '@gms/ui-state';
import {
  DisplayedEventsConfigurationEnum,
  EventFilterDropdownOptions,
  EventsColumn
} from '@gms/ui-state';
import Immutable from 'immutable';
/**
 * Event Table generic CellRendererParams type
 */
export type EventTableCellRendererParams<T> = CellRendererParams<
  EventRow,
  unknown,
  T,
  unknown,
  unknown
>;

/**
 * used to match the display strings to values in the site table column picker dropdown
 */
export const eventColumnDisplayStrings: Immutable.Map<EventsColumn, string> = Immutable.Map<
  EventsColumn,
  string
>([
  [EventsColumn.unsavedChanges, 'Unsaved changes'],
  [EventsColumn.conflict, 'Conflict'],
  [EventsColumn.time, 'Time'],
  [EventsColumn.timeUncertainty, 'Time Std Deviation (s)'],
  [EventsColumn.latitudeDegrees, 'Lat (°)'],
  [EventsColumn.longitudeDegrees, 'Lon (°)'],
  [EventsColumn.depthKm, 'Depth (km)'],
  [EventsColumn.depthUncertainty, 'Depth Std Deviation'],
  [EventsColumn.magnitudeMb, 'mb'],
  [EventsColumn.magnitudeMs, 'ms'],
  [EventsColumn.magnitudeMl, 'ml'],
  [EventsColumn.numberAssociated, '# Associated'],
  [EventsColumn.numberDefining, '# Defining'],
  [EventsColumn.observationsStandardDeviation, 'Sdobs'],
  [EventsColumn.coverageSemiMajorAxis, 'Coverage semi-major'],
  [EventsColumn.coverageSemiMinorAxis, 'Coverage semi-minor'],
  [EventsColumn.confidenceSemiMajorAxis, 'Confidence semi-major'],
  [EventsColumn.confidenceSemiMinorAxis, 'Confidence semi-minor'],
  [EventsColumn.region, 'Region'],
  [EventsColumn.activeAnalysts, 'Active analysts'],
  [EventsColumn.preferred, 'Preferred in stage'],
  [EventsColumn.status, 'Workflow status'],
  [EventsColumn.deleted, 'Deleted'],
  [EventsColumn.rejected, 'Rejected']
]);

/**
 * Used to match the display strings to values in the events dropdown.
 */
export const displayedEventsDropdownDisplayStrings: Immutable.Map<
  DisplayedEventsConfigurationEnum,
  string
> = Immutable.Map<DisplayedEventsConfigurationEnum, string>([
  [DisplayedEventsConfigurationEnum.edgeEventsBeforeInterval, EventFilterDropdownOptions.BEFORE],
  [DisplayedEventsConfigurationEnum.edgeEventsAfterInterval, EventFilterDropdownOptions.AFTER],
  [DisplayedEventsConfigurationEnum.eventsCompleted, EventFilterDropdownOptions.COMPLETE],
  [DisplayedEventsConfigurationEnum.eventsRemaining, EventFilterDropdownOptions.REMAINING],
  [DisplayedEventsConfigurationEnum.eventsConflict, EventFilterDropdownOptions.CONFLICTS],
  [DisplayedEventsConfigurationEnum.eventsDeleted, EventFilterDropdownOptions.DELETED],
  [DisplayedEventsConfigurationEnum.eventsRejected, EventFilterDropdownOptions.REJECTED]
]);

export const displayedEventsLabelStrings: Immutable.Map<
  DisplayedEventsConfigurationEnum,
  string
> = Immutable.Map<DisplayedEventsConfigurationEnum, string>([
  [DisplayedEventsConfigurationEnum.edgeEventsBeforeInterval, 'Edge Events'],
  [DisplayedEventsConfigurationEnum.eventsCompleted, 'Event Status']
]);

export const displayedEventsRenderDividers: Immutable.Map<
  DisplayedEventsConfigurationEnum,
  boolean
> = Immutable.Map<DisplayedEventsConfigurationEnum, boolean>([
  [DisplayedEventsConfigurationEnum.edgeEventsAfterInterval, true]
]);

/**
 * Enumerated
 *  - time types based on whether event or signal detection is within the interval or a before/after edge event
 *  - and Deleted or Rejected types
 */
export enum EventFilterOptions {
  BEFORE = 'Before',
  AFTER = 'After',
  INTERVAL = 'Interval',
  COMPLETE = 'Complete',
  DELETED = 'Deleted',
  CONFLICTS = 'Conflicts',
  REJECTED = 'Rejected'
}

export enum EventLifecycle {
  DELETED = 'Deleted',
  REJECTED = 'Rejected',
  ACTIVE = 'Active'
}

/**
 * Event row for event display
 */
export interface EventRow extends Row {
  readonly id: string;
  readonly unsavedChanges: boolean;
  readonly eventFilterOptions: EventFilterOptions[];
  readonly isOpen: boolean;
  readonly conflict: boolean;
  readonly time: ArrivalTime;
  readonly latitudeDegrees: number;
  readonly longitudeDegrees: number;
  readonly depthKm: EventTypes.Depth;
  readonly region: string;
  readonly numberAssociated: number;
  readonly numberDefining: number;
  readonly observationsStandardDeviation?: number;
  readonly confidenceSemiMajorAxis?: number;
  readonly confidenceSemiMinorAxis?: number;
  readonly coverageSemiMajorAxis?: number;
  readonly coverageSemiMinorAxis?: number;
  readonly magnitudeMb: number;
  readonly magnitudeMs: number;
  readonly magnitudeMl: number;
  readonly activeAnalysts: string[];
  readonly preferred: boolean;
  readonly status: EventTypes.EventStatus;
  readonly deleted: boolean;
  readonly rejected: boolean;
  readonly isActionTarget: boolean;
  readonly isUnqualifiedActionTarget: boolean;
}
