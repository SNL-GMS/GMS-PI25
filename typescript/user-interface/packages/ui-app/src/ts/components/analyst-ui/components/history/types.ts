import type { HistoryAction } from '@gms/ui-state';

export interface HistoryMeta {
  readonly type: HistoryAction;
  readonly isApplied: boolean;
  readonly isIncluded: boolean;
  readonly isAssociated: boolean;
  readonly isAssociatedToOther: boolean;
  readonly isUnAssociated: boolean;
  readonly isRelatedToEvent: boolean;
  readonly isCompleted: boolean;
  readonly isConflictCreated: boolean;
  readonly isConflictResolved: boolean;
  readonly isDeletion: boolean;
  readonly isRejection: boolean;
}

export interface HistoryChange extends HistoryMeta {
  readonly id: string;
  readonly historyId: string;
  readonly label: string;
  readonly description: string;
}

export interface History extends HistoryMeta {
  readonly id: string;
  readonly historyId: string;
  readonly label: string;
  readonly description: string;
  readonly changes: HistoryChange[];
}

export interface HistoryPointer {
  readonly index: number;
}
