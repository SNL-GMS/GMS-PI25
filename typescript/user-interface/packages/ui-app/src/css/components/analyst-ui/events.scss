.events-status-bar {
  display: flex;
  flex: 0 0 auto;
  justify-content: space-between;
  align-items: center;
}

.events-status-bar__visibility-menu-button {
  padding-left: 0px; // Override for default to make flush left
}

.events-status-bar__event-counters {
  display: flex;
  flex: 0 0 auto;
  justify-content: space-between;
  & > * {
    padding-right: 0.5rem; // All direct children have right padding to enforce spacing
  }
}

.events-status-bar__mark-button {
  display: flex;
  flex: 1 0 auto;
  justify-content: flex-end;
}

.events__mark-complete-button {
  height: 20px;
  min-height: 23px !important;
  min-width: 0px !important;
  position: relative;
  top: -1px;
}

.events__conflict-marker {
  height: 36px;
  flex: auto;
  display: flex;
  justify-content: center;
}

.event-panel {
  display: flex;
  height: 100%;
  width: 100%;
  flex-direction: column;
  overflow-y: auto;
  overflow-x: hidden;
}

.event-table {
  height: 100%;
  width: 100%;
}

.events-display-window {
  @include flexColumn;
  width: 100%;
  height: 100%;
  .toolbar {
    margin-bottom: 0.5em;
  }
}

.table-cell .table-cell__value--complete {
  color: $analyst-complete;
}

.ag-theme-dark .ag-row-even.edge-event-row .table-cell {
  background-color: var(--gms-background, $gms-background);
  filter: brightness(0.5);
}

.ag-theme-dark .ag-row-odd.edge-event-row .table-cell {
  background-color: var(--gms-background, $gms-background);
  filter: brightness(0.5);
}

.ag-theme-dark .ag-row-even.open-event-row .table-cell {
  background-color: var(--analyst-open-event, $analyst-open-event);
  .table-cell__value {
    color: var(--open-associated-text-color, $gms-main);
  }
}

.ag-theme-dark .ag-row-odd.open-event-row .table-cell {
  background-color: var(--analyst-open-event, $analyst-open-event);
  .table-cell__value,
  svg {
    color: var(--open-associated-text-color, $gms-main);
  }
}

.bp4-popover2-content-icon-margin-left {
  margin-left: 15px;
}

.bp4-popover2-content-dialog-container-left-right-padding {
  padding-left: 1.5em;
  padding-right: 1.5em;
}

@container (max-width: 900px) {
  .event-panel {
    .counter__count--prefix {
      display: none;
    }
    .counter__entry + .counter__entry {
      margin-left: 1rem;
    }
    .counter__count {
      margin-left: 0.25rem;
    }
  }
}

@container (max-width: 650px) {
  .event-panel {
    .counter__entry + .counter__entry {
      margin-left: 0.5rem;
    }
    .counter__label {
      display: none;
    }
  }
}
