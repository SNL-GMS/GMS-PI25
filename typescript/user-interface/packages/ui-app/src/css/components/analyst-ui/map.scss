@import '../../gms-fonts.scss';

.map {
  width: 100%;
  height: 100%;
  padding: 0.5rem;
}
.map__status-online-indicator {
  position: absolute;
  bottom: 0.8rem;
  right: 1rem;
  color: var(--gms-main, $gms-main);
  .map__status-online-indicator__online-mode {
    color: var(--gms-main, $gms-main);
    text-shadow: -1px 0 var(--gms-recessed, $gms-recessed), 0 1px var(--gms-recessed, $gms-recessed),
      1px 0 var(--gms-recessed, $gms-recessed), 0 -1px var(--gms-recessed, $gms-recessed);
  }
  .map__status-online-indicator__offline-mode {
    color: $gms-strong-warning;
    text-shadow: -1px 0 var(--gms-recessed, $gms-recessed), 0 1px var(--gms-recessed, $gms-recessed),
      1px 0 var(--gms-recessed, $gms-recessed), 0 -1px var(--gms-recessed, $gms-recessed);
  }
}
.map__column {
  display: flex;
  height: 100%;
  flex-direction: column;
}
.map__row {
  flex: 1 1 auto;
  display: flex;
}
.map-controls {
  flex: 0 0 auto;
  display: flex;
  align-items: center;
  padding-left: 0.5rem;
}
.map-controls__label {
  margin-bottom: 0rem !important;
  margin-left: 1rem !important;
}
.map-layer-toggles {
  flex: 0 0 auto;
  width: 130px;
  display: flex;
  flex-direction: column;
  background-color: var(--gms-prominent-background, $gms-prominent-background);
}
.map__inner-column {
  flex: 1 1 auto;
  border-left: 1px solid var(--gms-background, $gms-background);
  display: flex;
  flex-direction: column;
}
.map__rendering-wrapper-1 {
  flex: 1 1 auto;
  position: relative;
}

.map__layer-button {
  position: absolute;
  z-index: 10;
  top: 5px; //to align with cesium toolbar buttons
  left: 5px;
}

.map-time {
  font-family: $gms-mono;
}

.ian-map-gl-container {
  height: 100%;
  padding: 0;
}

.layer-panel-content {
  &__container {
    margin-top: -28px;
    margin-left: 10px;
    position: absolute;
    background-color: var(--gms-popover-background, $gms-popover-background);
  }
  &--header {
    padding: 8px 8px 8px 12px;
    border-bottom: solid 1px;
  }
  &--checkbox-list {
    margin-left: 8px;
  }
}

// Disabling Columbus view
.cesium-sceneModePicker-dropDown-icon[data-bind*='COLUMBUS_VIEW'] {
  display: none;
}

.layer-panel-content--header .bp4-heading {
  color: var(--gms-prominent, $gms-prominent) !important;
}

.menu-item-create-event {
  &__lat {
    font-family: $gms-mono;
  }
  &__lon {
    padding-left: 1px;
    font-family: $gms-mono;
  }
}
