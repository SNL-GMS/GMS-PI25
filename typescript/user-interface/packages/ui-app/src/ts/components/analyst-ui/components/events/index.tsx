import { IanDisplays, toDisplayTitle } from '@gms/common-model/lib/displays/types';
import { nonIdealStateWithSpinner } from '@gms/ui-core-components';
import React from 'react';

import type { EventsComponentProps } from './events-component';

const EventsLazy = React.lazy(async () =>
  import(/* webpackChunkName: 'ui-app-events' */ './events-component').then(module => ({
    default: module.EventsComponent
  }))
);

export function Events(props: EventsComponentProps) {
  return (
    <React.Suspense
      fallback={nonIdealStateWithSpinner(toDisplayTitle(IanDisplays.EVENTS), 'Loading Display')}
    >
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <EventsLazy {...props} />
    </React.Suspense>
  );
}
