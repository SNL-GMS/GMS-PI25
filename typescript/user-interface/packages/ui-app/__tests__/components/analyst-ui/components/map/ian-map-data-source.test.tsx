import { SignalDetectionTypes } from '@gms/common-model';
import {
  eventData,
  processingAnalystConfigurationData,
  signalDetectionsData
} from '@gms/common-model/__tests__/__data__';
import type { Entity } from 'cesium';
import cloneDeep from 'lodash/cloneDeep';

import { createSignalDetectionEntities } from '~analyst-ui/components/map/create-ian-entities';

import { EventFilterOptions } from '../../../../../src/ts/components/analyst-ui/components/events/types';
import {
  sdOnMouseEnterHandler,
  sdOnMouseLeaveHandler
} from '../../../../../src/ts/components/analyst-ui/components/map/ian-map-utils';

describe('ian map data source', () => {
  const associatedEvent = cloneDeep(eventData);
  signalDetectionsData.forEach(sd => {
    associatedEvent.eventHypotheses[0].associatedSignalDetectionHypotheses.push(
      sd.signalDetectionHypotheses[0]
    );
  });

  const entities: Entity[] = createSignalDetectionEntities(
    {
      signalDetections: signalDetectionsData,
      selectedSdIds: [],
      edgeSDType: EventFilterOptions.INTERVAL,
      status: SignalDetectionTypes.SignalDetectionStatus.OTHER_ASSOCIATED,
      sdVisibility: true,
      signalDetectionLengthMeters: null
    },
    [],
    [],
    processingAnalystConfigurationData.uiThemes[0],
    []
  );

  test('sdOnMouseEnterHandler', () => {
    const result = sdOnMouseEnterHandler({} as any, entities[0]);
    expect(result).toMatchSnapshot();
  });

  test('sdOnMouseLeaveHandler', () => {
    const result = sdOnMouseLeaveHandler({} as any, entities[0]);
    expect(result).toMatchSnapshot();
  });
});
