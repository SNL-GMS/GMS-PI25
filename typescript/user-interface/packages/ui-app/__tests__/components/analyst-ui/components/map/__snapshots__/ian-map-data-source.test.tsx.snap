// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ian map data source sdOnMouseEnterHandler 1`] = `undefined`;

exports[`ian map data source sdOnMouseLeaveHandler 1`] = `undefined`;
